provider.go

```diff
@@ -38,25 +40,88 @@ func Provider(configfn schema.ConfigureFunc) terraform.ResourceProvider {
 	}
 }
 
+// Get the default value from the first function that returns one without an error
+func firstOf(key string, dvs ...schema.SchemaDefaultFunc) schema.SchemaDefaultFunc {
+	return func() (interface{}, error) {
+		for _, dv := range dvs {
+			if cfg, err := dv(); cfg != nil && err == nil {
+				return cfg, nil
+			}
+		}
+
+		return nil, fmt.Errorf("Unable to retrieve default for '%s' from anywhere", key)
+	}
+}
+
+// Look for default values in the BMC configuration file
+func fileBasedConfig(file string, key string) schema.SchemaDefaultFunc {
+	return func() (interface{}, error) {
+		cfg, err := ini.InsensitiveLoad(file)
+		if cfg != nil && err == nil {
+			sec, err := cfg.GetSection("")
```

Comment:
This looks like it will not work with a typical ini file that has named sections like 
[DEFAULT]
...
[USER1]
...